### PR TITLE
Add getAssetChanges — a new JSON-RPC endpoint for incremental asset indexing

### PR DESCRIPTION
**Pull request by joenoon**
Maintaining a continuously up-to-date index of all NFTs across owners, collections, and assets requires incremental sync. The DAS API stores slot_updated on every asset but has no way to expose it as a change feed.

We investigated building this on searchAssets but it's not viable:
- Cursor pagination requires sortBy: Id, so you can't cursor-paginate while sorting by slot
- slot_updated is never returned in the response — no per-item checkpoint marker
- last_indexed_slot in searchAssets is a global watermark, not per-asset

A dedicated endpoint avoids these constraints without risk to existing searchAssets clients.

### What it does

Returns a forward-paginated feed of assets ordered by slot_updated, with per-item fields sufficient to reduce the need for many getAsset or getAssetBatch followup requests:

Request:
```
{
  "method": "getAssetChanges",
  "params": {
    "afterSlot": 250000000,
    "limit": 1000,
    "after": "<cursor>"
  }
}
```

Response:
```
{
  "currentSlot": 250731042,
  "items": [
    {
      "id": "Abc...xyz",
      "slotUpdated": 250000183,
      "owner": "Wallet...1",
      "delegate": null,
      "burnt": false,
      "collection": "Col...mint",
      "metadataUrl": "https://arweave.net/abc"
    }
  ],
  "after": "<next_cursor>"
}
```

- currentSlot — max indexed slot, so clients know when they've reached head
- slotUpdated — per-item checkpoint marker
- owner / delegate / burnt — ownership and listing state
- collection — from asset_grouping via LEFT JOIN LATERAL
- metadataUrl — from asset_data via LEFT JOIN

Cursor pagination uses composite keyset (slot_updated, id) encoded as bs58. afterSlot is exclusive (strictly >) to enable clean checkpoint semantics. A migration adds a composite index on (slot_updated, id) to support this.

### Implementation notes

- Raw SQL with LEFT JOIN LATERAL follows established patterns in dao/scopes/asset.rs and dapi/change_logs.rs
- verified = true OR verified IS NULL matches the backward-compatibility pattern used in every collection query (older indexer versions didn't write the verified flag)
- encode_change_cursor / decode_change_cursor are public functions in digital_asset_types for testability

### Tests

- 10 unit tests: response serialization (camelCase keys, skip_serializing_if), cursor encode/decode round-trips, edge cases (zero slot, max slot, minimum valid size), invalid input rejection (too short, bad base58)
- 3 integration tests: basic query with seeded NFT, cursor pagination across pages, empty database behavior

## UPDATES — commit 2: assetTypes filter, type response field, creator field

The initial endpoint returns all asset types indiscriminately. Since the asset table indexes everything — NFTs, fungible tokens, unknown types — NFT-focused consumers get flooded with irrelevant changes and vice versa.

This commit adds:

- **assetTypes** (required request parameter) — filters by simplified category: `NFT`, `FungibleAsset`, `FungibleToken`, `Custom`. Maps to `specification_asset_class` values in the DB via a parameterized `IN (...)` clause with `::text` cast.
- **type** (response field) — the precise `Interface` value (`V1_NFT`, `ProgrammableNFT`, `MplCoreAsset`, etc.), derived from `specification_version` + `specification_asset_class` using the existing `Interface::from` impl.
- **creator** (response field) — first verified creator via a new `LEFT JOIN LATERAL` on `asset_creators`, reducing getAsset followup calls further.

Request:
```
{
  "method": "getAssetChanges",
  "params": {
    "assetTypes": ["NFT"],
    "afterSlot": 250000000,
    "limit": 1000
  }
}
```

Response items now include:
```
{
  "id": "Abc...xyz",
  "type": "ProgrammableNFT",
  "slotUpdated": 250000183,
  "owner": "Wallet...1",
  "creator": "Creator...1",
  ...
}
```

`assetTypes` is required and validated non-empty. Categories are deduplicated before expansion to DB values.

### Tests

- 4 new unit tests: AssetCategory serialization/deserialization round-trips, to_asset_classes mapping, type field always-present assertion
- Existing unit and integration tests updated for new required field and response shape

### CODE REVIEW

File: digital_asset_types/src/dao/scopes/asset.rs [hunk 1]
```diff
@@ -816,3 +817,108 @@ async fn find_tokens(
...
+                 AND (ag.verified = true OR ag.verified IS NULL)
+               LIMIT 1
+           ) lat ON true
+           LEFT JOIN LATERAL (
+               SELECT ac.creator
+               FROM asset_creators ac
+               WHERE ac.asset_id = a.id AND ac.verified = true
+               ORDER BY ac.position ASC
+               LIMIT 1
+           ) lat_cr ON true
```
[the remark this answers is not in the document]
Comment (joenoon) [reply digital_asset_types/src/dao/scopes/asset.rs:893]: I don't think this applies here. asset_creators.verified is a non-nullable bool column (unlike asset_grouping.verified which is Option<bool>), so the IS NULL guard would be dead code. And adding slot_updated DESC before position ASC would change the semantics to pick the most-recently-updated creator rather than the first-by-position, which is inconsistent with how creator ordering works elsewhere in the codebase (e.g., get_related_for_assets and the single-asset query both order purely by position ASC).  But please lmk.
